### PR TITLE
NEXT-36 Introduce unit test for NEXT

### PR DESCRIPTION
**Pull request by xzfc**
Added tests for:
* `animated-value.js`
* `map-size.js`
* `misc.js`

Tests are not added for:
* `ball-view.js` - uses pixi.
* `connector.js` - uses XMLHttpRequest. Can be mocked however.
* `constants.js` - no need to test.
* `controller.js` - uses pixi.
* `main.js` - no need to test.
* `pointer.js` - relies on `Viewer` object that passed into it; also mouse events.
* `viewer.js` - uses pixi.

### CODE REVIEW

File: test/.jshintrc [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "globalstrict": true,
+  "browser": true,
+  "node": true,
+  "expr": true,
+  "esnext": true,
+  "globals": {
+    "describe"   : false,
```
Comment (heroandtn3) [inline test/.jshintrc]: "mocha: true" is enough

File: test/animated-value.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+const AnimatedValue = require('../client/js/animated-value');
+const assert = require('assert');
```
Comment (heroandtn3) [inline test/animated-value.js]: Please use chai.js http://chaijs.com/api/bdd/ and use expect syntax like this:

`expect(true).to.be.true`

File: test/animated-value.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+describe('AnimatedValue', function () {
+  describe('constructor', function () {
+    it('should set initial value', function () {
+      assert.equal(new AnimatedValue(10).get(), 10);
+      assert.equal(new AnimatedValue(20).get(), 20);
+    });
+  });
+
+  describe('write', function () {
```
Comment (heroandtn3) [inline test/animated-value.js]: "write method"

File: test/animated-value.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    it('should set value immediately', function () {
+      let v = new AnimatedValue(10);
+      v.write(20);
+      assert.equal(v.get(), 20);
+    });
+  });
+
+  describe('set', function () {
```
Comment (heroandtn3) [inline test/animated-value.js]: "set method"

File: test/map-size.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+const MapSize = require('../client/js/map-size');
+const assert = require('chai').assert;
+
+describe('MapSize', function () {
+  it('should have proper centerX',
+     () => assert.equal(new MapSize(-10, -20, 30, 50).centerX(), 10));
```
Comment (heroandtn3) [inline test/map-size.js]: Please use {} for function because we may add more statements later.

**Comment by heroandtn3** [comment]
@xzfc please update your PR

**Comment by xzfc** [comment]
Updated.

**Pull request merged**